### ISSUE TITLE
When model is invalid, model.set('attr', 'value') will not stick even if the property is valid

### ISSUE THREAD

**Issue by davisford** [thedersen/backbone.validation#32]
TL;DR: if a model's initial state after creation is invalid, then you can't set other attributes on the model until the offending attribute is fixed.  calls to this.model.set('attr', 'value') are ignored.

Here's the long version:  consider a model like this:

``` javascript
ED.Models.Time = Backbone.Model.extend({
    defaults: {
      connections: []
    },
    validation: {
      time: { required: true, msg: "You must set a time value" },
      connections: function (val) {
        if (!val || val.length === 0) { return "You must connect this to something"; }
      }
    }
  });
```

So, initially, connections is an empty array, and that means it is invalid according to the validation rules.  The other validation rules are for properties that are fed values through an html form.  They are required, but since they do not have a default value, validation skips them until you try to set them.

When I go to fill out the form, I have change events bound to all the form fields, and I'll do something like this:

``` javascript
timeChanged: function () {
      this.model.set('time', this.$('.ed-widget-timepicker')[0].value);
      console.log(JSON.stringify(this.model.toJSON()));
    }
```

So, initially, the view's model is invalid b/c it does not contain anything in the connections array.  When I go to set the time, and trigger the timeChanged function above, you'll see that I try to set the time property on the model, but it does not take...validation is executed upon set, and the value is not set in the model b/c the model is invalid.

If I then add something to the connections array, so that attribute is now valid, and then try to set the time attribute, it does set the attribute properly on the model.

This is unexpected.  I would have assumed that I could still set the time property even if the connections is invalid.
